### PR TITLE
CFY-2108 REST service

### PR DESCRIPTION
**Pull request by msoczewka**

### CODE REVIEW

File: cloudify_hostpool/rest/backend/acquire.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+                acquire = True
+                break
+        finally:
+            if not acquire:
+                db.update_host(host['global_id'], {'reserved': False})
+    # No host acquired
+    else:
+        return None
+    host = db.update_host(host['global_id'],
+                          {'host': str(uuid.uuid4()),
```
Comment (iliapolo) [inline cloudify_hostpool/rest/backend/acquire.py]: shouldn't this be host_id?

File: cloudify_hostpool/rest/backend/acquire.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+                           'reserved': False})
+    return host
+
+
+def _aquisition_gen(db):
+    for alive in True, False:
+        for host in db.get_host(reserved=False, alive=alive):
+            if host.get('host') is None:
```
Comment (iliapolo) [inline cloudify_hostpool/rest/backend/acquire.py]: same here. host_id?

File: cloudify_hostpool/rest/backend/acquire.py (hunk 1 above)
Comment (szpotona) [inline cloudify_hostpool/rest/backend/acquire.py]: Yes, indeed. This is already changed in branch "Filters".

File: cloudify_hostpool/rest/backend/acquire.py (hunk 2 above)
Comment (szpotona) [inline cloudify_hostpool/rest/backend/acquire.py]: Yes, indeed. This is already changed in branch "Filters" so you can safely merge this branch and "Filters" together.

File: cloudify_hostpool/rest/service.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+from flask import Flask
+from flask import jsonify
+from flask_restful import Api
+
+from cloudify_hostpool import exceptions
+from cloudify_hostpool.rest.backend import rest_backend
+
+app = Flask(__name__)
+api = Api(app)
+backend = rest_backend.RestBackend('test.yaml', 'test.db')
```
Comment (iliapolo) [inline cloudify_hostpool/rest/service.py]: left overs from test?

Comment (szpotona) [inline cloudify_hostpool/rest/service.py]: Nope. For now I haven't found any good manner to pass those arguments from command calling gunicorn - so I left them hardcoded.

Comment (iliapolo) [inline cloudify_hostpool/rest/service.py]: im not sure i follow. it seems like we are missing an important feature...like actually using a configuration file on the machine...

File: cloudify_hostpool/rest/backend/acquire.py [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+        return None
+    host = db.update_host(host['global_id'],
+                          {'host_id': str(uuid.uuid4()),
+                           'reserved': False})
+    return host
+
+
+def _aquisition_gen(db):
+    for alive in True, False:
+        for host in db.get_host(reserved=False, alive=alive):
```
Comment (iliapolo) [inline cloudify_hostpool/rest/backend/acquire.py]: this should be db.get_hosts (not db.get_host)

File: cloudify_hostpool/rest/service.py (hunk 3 above)
Comment (szpotona) [inline cloudify_hostpool/rest/service.py]: We're not missing it, it just should be named 'test.yaml' and be in folder where we run gunicorn from. DB will handle itself.

Comment (iliapolo) [inline cloudify_hostpool/rest/service.py]: oh ok. so i think that for now lets at least change the name of the file...test.yaml is kind of strage.
how about host-pool.yaml? db as well

File: cloudify_hostpool/rest/backend/acquire.py (hunk 4 above)
Comment (msoczewka) [inline cloudify_hostpool/rest/backend/acquire.py]: Good point.

File: cloudify_hostpool/rest/service.py (hunk 3 above)
Comment (msoczewka) [inline cloudify_hostpool/rest/service.py]: I've changed the names.

File: cloudify_hostpool/rest/backend/acquire.py [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+from cloudify_hostpool.backend import scan
+
+
+def acquire(db):
+    '''"Creates" (acquires) a host.
+
+    :return: host dict if successful, None otherwise.
+    :param db: host database.
+    '''
+    for host in _aquisition_gen(db):
```
Comment (iliapolo) [inline cloudify_hostpool/rest/backend/acquire.py]: i makes it a bit difficult to understand when you use an iteration variable outside of the iteration loop.
i think a better construct would just be returning the host from inside the loop and not using break statements.

File: cloudify_hostpool/rest/backend/acquire.py [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+    results = scan.scan([(address, port)])
+    is_alive = results[address, port]
+    while True:
+        try:
+            hst = db.update_host(host['global_id'], {'alive': is_alive})
+        except exceptions.DBLockedError:
+            pass
+        else:
+            break
+    if hst is None:
```
Comment (iliapolo) [inline cloudify_hostpool/rest/backend/acquire.py]: you are using a variable that was only defined in a try block. this is a bit shaky, why not just put the return statement inside the try?

File: cloudify_hostpool/rest/backend/acquire.py [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+        try:
+            if not db.reserve_host(host['global_id']):
+                continue
+            host = _check_if_alive(db, host)
+            if host['alive']:
+                acquire = True
+                break
+        finally:
+            if not acquire:
+                db.update_host(host['global_id'], {'reserved': False})
```
Comment (iliapolo) [inline cloudify_hostpool/rest/backend/acquire.py]: this call to update_host can also raise an exception

**Comment by msoczewka** [comment]
ad. `db.update_host()`: #28

**Pull request merged**